### PR TITLE
Build with oraclejdk11

### LINKED ISSUE #763: Support Java 11
I'm working to build and test SpotBugs on OpenJDK 11, here is [a branch](https://github.com/spotbugs/spotbugs/compare/build-on-jdk11) for this trial so you can see detail.

Problem is that, [Java 11 does not provide methods like `System.runFinalizersOnExit()` and `Thread#destroy()`](https://travis-ci.org/spotbugs/spotbugs/jobs/433389109#L2201). So we cannot test SpotBugs on OpenJDK 11 for now.

In my opinion, we can remove `classesJava7` task in `spotbugsTestCases/build.gradle` since we've removed Java 7 support. If you have different opinion, let's discuss at here.

### PR DESCRIPTION
**Pull request by KengoTODA**
This PR closes #763. It contains following changes:

1. Currently we have test code that needs JDK7 to compile, however [spotbugs does not support scanning Java7](https://github.com/spotbugs/spotbugs/issues/19). And JDK11 cannot compile with `target = 1.7` so I removed code under `spotbugsTestCases/src/java7`.
2. `javax.annotation` has been removed in JDK11, so added dependency on it explicitly.
3. JDK11 removed some API like `Thread#destroy()`, so move test depending on them into `spotbugsTestCases/src/java8` and skip them when we build with JDK11.
4. Use `oraclejdk11` to build. It makes test failed (see #756) so [allow failure](https://docs.travis-ci.com/user/customizing-the-build/#rows-that-are-allowed-to-fail) as temporal solution.

### CODE REVIEW

**Comment by KengoTODA** [comment]
@spotbugs/everyone please review this PR, thank you!

File: .travis.yml [hunk 1]
```diff
@@ -49,7 +51,7 @@ install:
 script:
   - jdk_switcher use oraclejdk8
   - ./gradlew classes testClasses -S
-  - jdk_switcher use $TRAVIS_JDK_VERSION
+  - if [[ $TRAVIS_JDK_VERSION == "oraclejdk11" ]]; then export JAVA_HOME=$HOME/oraclejdk11; fi
```
Comment (ThrawnCA) [inline .travis.yml]: What provides this JDK?

Comment (KengoTODA) [reply .travis.yml]: TravisCI installs it by [install-jdk.sh](https://github.com/sormuras/bach#install-jdksh).

Here is [official documentation about how to switch JDK 10 and up](https://docs.travis-ci.com/user/languages/java/#switching-jdks-to-java-10-and-up-within-one-job).

Comment (iloveeclipse) [reply .travis.yml]: Just wondering that we first switch to jdk 8 but next line check if the jdk version is 11? How is this possible?

Comment (KengoTODA) [reply .travis.yml]: 9 and 10 have already achieved their EOL. Recently Java changed their release policy.

Comment (iloveeclipse) [reply .travis.yml]: I was not wondering where 9 and 10 are, I was trying to understand why the script switches to java 8 but a line after tests if it will be 11? I simply can't understand this part of the script. If I switch to 8 I expect 8 to be used, not 11.

Comment (KengoTODA) [reply .travis.yml]: By this change, we'll have two build jobs [like this](https://travis-ci.org/spotbugs/spotbugs/builds/448196622).

One will **build and test** spotbugs with JDK8,  
another will **build** spotbugs with JDK8 and **test** it with JDK11.

Its background is:

1. We distribute binary built by JDK8, so build part should always use JDK8 (realized by `jdk_switcher use oraclejdk8` in `.travis.yml`)
2. Target classes to analyse can be compiled by different JDK in users' environment, and JVM that runs SpotBugs also can be different JDK. So test part should have several patterns. (realized by `if [[ $TRAVIS_JDK_VERSION == "oraclejdk11" ]]; then ...` in `.travis.yml`)

In other words, our build matrix is like this:

|#|JDK to compile|JDK to run test|
|-|----|----|
|1|JDK8|JDK8|
|2|JDK8|JDK11|

> If I switch to 8 I expect 8 to be used, not 11.

That's right. For the 1st pattern, we never use JDK11. We use it only for the 2nd pattern.

Comment (iloveeclipse) [reply .travis.yml]: Thanks Kengo. This now makes sense, but I beat next time I will see this code I will ask the same question again :-).

File: spotbugsTestCases/src/java7/sfBugs/Bug3458246.java [hunk 2]
```diff
@@ -1,37 +0,0 @@
-package sfBugs;
```
Comment (iloveeclipse) [inline spotbugsTestCases/src/java7/sfBugs/Bug3458246.java]: The bug is https://sourceforge.net/p/findbugs/bugs/1003/, shouldn't we move this to Java 8 directory instead of deleting it?

Comment (KengoTODA) [reply spotbugsTestCases/src/java7/sfBugs/Bug3458246.java]: OK I will.

**Comment by KengoTODA** [comment]
Sorry but please review this PR again, thanks @ThrawnCA !

File: spotbugsTestCases/src/java/sfBugs/Bug3458246.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package sfBugs;
+
+import java.util.logging.Logger;
```
Comment (ThrawnCA) [inline spotbugsTestCases/src/java/sfBugs/Bug3458246.java]: Is SLF4J fixed yet? There's supposed to be a Gradle fix.

Comment (KengoTODA) [reply spotbugsTestCases/src/java/sfBugs/Bug3458246.java]: It is old test code, not related with that trouble.
